src/coreComponents/dataRepository/GroupContext.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file GroupContext.cpp
+ */
+
+#include "GroupContext.hpp"
+
+namespace geos
+{
+namespace dataRepository
+{
+
+
+GroupContext::GroupContext( Group & group, string const & objectName ):
+  DataContext( objectName, false ),
+  m_group( group )
+{}
+GroupContext::GroupContext( Group & group ):
+  GroupContext( group, group.getName() )
+{}
+
+string GroupContext::toString() const
+{
+  string path;
+  bool foundNearestLine = false;
+  for( Group const * parentGroup = &m_group; parentGroup->hasParent(); parentGroup = &parentGroup->getParent() )
+  {
+    if( !foundNearestLine && parentGroup->getDataContext().isDataFileContext() )
+    {
+      DataFileContext const & parentContext =
+        dynamic_cast< DataFileContext const & >( parentGroup->getDataContext() );
+      if( parentContext.getLine() != xmlWrapper::xmlDocument::npos )
+      {
+        path.insert( 0, '/' + parentGroup->getName() + '(' +
+                     splitPath( parentContext.getFilePath() ).second +
+                     ",l." + std::to_string( parentContext.getLine() ) + ')' );
+        foundNearestLine=true;
+      }
+      else
+      {
+        path.insert( 0, '/' + parentGroup->getName() );
+      }
```

Comment:
Is it possible to change the architecture such that you can ask `parentContext.getDescription()` (or anything) that returns what it needs, depending on its real type, without having to down cast to the real type? Something like
```c++
for( Group const * parentGroup = &m_group; parentGroup->hasParent(); parentGroup = &parentGroup->getParent() )
{
    path.insert( 0, '/' + parentGroup->getName()  + parentContext.getDescription() );
}
```
whatever the real type of `DataContext` is?

That way, we could get rid of the `isDataFileContext` and the downcasting technique.
I could spot usages in the `tests` which we surely can adapt.
And another one in the `WrapperContext`, which we can surely adapt in the same way.

Can you investigate this?